addons/mail/static/src/core/common/record.js

```diff
@@ -51,6 +52,12 @@ function updateAttr(record, fieldName, value) {
     // ensure each field write goes through the proxy exactly once to trigger reactives
     const targetRecord = record._proxyUsed.has(fieldName) ? record : record._proxy;
     let shouldChange = record[fieldName] !== value;
+    if (fieldDefinition?.type === "datetime" && value) {
+        if (!(value instanceof luxon.DateTime)) {
+            value = deserializeDateTime(value);
+        }
+        shouldChange = !record[fieldName]?.equals(value);
```

Comment:
`!` doesn't play nice with `?`
Coming from falsy, it should always change if there is a value.

Reply:
I guess it's fine, but it's very hard to read and understand.